### ISSUE TITLE
Building wine-tkg error

### ISSUE THREAD

**Issue by Denkevich** [Frogging-Family/wine-tkg-git#375]
Hi TGK!
Linux Mint 20.1 Cinnamon 64bit
Tried building wine-tkg 6.8 in git with: gcc9, gcc10, gcc11.
Packages installed: xorg-dev libx11-dev libgstreamer1.0-dev libgstreamer-plugins-base1.0-dev build-essential manpages-dev software-properties-common xserver-xorg-dev libfreetype6-dev libfontconfig1-dev libglu1-mesa-dev libosmesa6-dev mingw-w64.
I get an error:
gcc -m64 -c -o dlls/winex11.drv/vulkan.o ../wine-mirror-git/dlls/winex11.drv/vulkan.c -Idlls/winex11.drv \
  -I../wine-mirror-git/dlls/winex11.drv -Iinclude -I../wine-mirror-git/include -D__WINESRC__ \
  -D_REENTRANT -fPIC -fasynchronous-unwind-tables -Wall -pipe -fcf-protection=none \
  -fno-stack-protector -fno-strict-aliasing -Wdeclaration-after-statement -Wempty-body \
  -Wignored-qualifiers -Winit-self -Wno-packed-not-aligned -Wshift-overflow=2 -Wstrict-prototypes \
  -Wtype-limits -Wunused-but-set-parameter -Wvla -Wwrite-strings -Wpointer-arith -Wlogical-op \
  -O2 -ftree-vectorize -U_FORTIFY_SOURCE -D_FORTIFY_SOURCE=0
../wine-mirror-git/dlls/winex11.drv/vulkan.c:878:6: error: conflicting types for ‘resize_vk_surfaces’; have ‘void(struct HWND__ *, Window,  int,  XWindowChanges)’ {aka ‘void(struct HWND__ *, long unsigned int,  int,  XWindowChanges)’}
  878 | void resize_vk_surfaces(HWND hwnd, Window active, int mask, XWindowChanges changes)
      |      ^~~~~~~~~~~~~~~~~~
In file included from ../wine-mirror-git/dlls/winex11.drv/vulkan.c:34:
../wine-mirror-git/dlls/winex11.drv/x11drv.h:592:13: note: previous declaration of ‘resize_vk_surfaces’ with type ‘void(struct HWND__ *, Window,  int,  XWindowChanges *)’ {aka ‘void(struct HWND__ *, long unsigned int,  int,  XWindowChanges *)’}
  592 | extern void resize_vk_surfaces( HWND hwnd, Window active, int mask, XWindowChanges *changes ) DECLSPEC_HIDDEN;
      |             ^~~~~~~~~~~~~~~~~~
make: *** [Makefile:224767: dlls/winex11.drv/vulkan.o] Ошибка 1
make: *** Ожидание завершения заданий…
 -> Removed BIG_UGLY_FROGMINER - Ribbit
 -> Removed Proton-tkg token - Valve Ribbit
 -> exit cleanup done
 -> Compilation time for 64-bit wine:

How to fix it ?

**Comment by Tk-Glitch** [comment]
Are you using the default configuration? If you aren't, please send your prepare.log over.

**Comment by Denkevich** [comment]
Yes, everything is by default, changed only the parameter nomakepkg_dep_resolution_distro="debuntu"

**Comment by Tk-Glitch** [comment]
Sadly I cannot reproduce the issue on Arch, which tends to point to some distro-specific issue. I'll try to find the time to install mint in a vm to find out what's going on.

**Comment by Denkevich** [comment]
Thank you very much! I will wait.

**Comment by pa5hgan** [comment]
Same issue on Gentoo unstable:

gcc -m32 -c -o dlls/winex11.drv/x11drv_main.o ../wine-mirror-git/dlls/winex11.drv/x11drv_main.c -Idlls/winex11.drv \
  -I../wine-mirror-git/dlls/winex11.drv -Iinclude -I../wine-mirror-git/include -D__WINESRC__ \
  -D_REENTRANT -fno-PIC -fasynchronous-unwind-tables -Wall -pipe -fcf-protection=none \
  -fno-stack-protector -fno-strict-aliasing -Wdeclaration-after-statement -Wempty-body \
  -Wignored-qualifiers -Winit-self -Wno-packed-not-aligned -Wshift-overflow=2 -Wstrict-prototypes \
  -Wtype-limits -Wunused-but-set-parameter -Wvla -Wwrite-strings -Wpointer-arith -Wlogical-op \
  -fno-omit-frame-pointer -O2 -ftree-vectorize -U_FORTIFY_SOURCE -D_FORTIFY_SOURCE=0
../wine-mirror-git/dlls/winex11.drv/vulkan.c:878:6: ошибка: conflicting types for «resize_vk_surfaces»; have «void(struct HWND__ *, Window,  int,  XWindowChanges)» {aka «void(struct HWND__ *, long unsigned int,  int,  XWindowChanges)»}
  878 | void resize_vk_surfaces(HWND hwnd, Window active, int mask, XWindowChanges changes)
      |      ^~~~~~~~~~~~~~~~~~
In file included from ../wine-mirror-git/dlls/winex11.drv/vulkan.c:34:
../wine-mirror-git/dlls/winex11.drv/x11drv.h:592:13: замечание: previous declaration of «resize_vk_surfaces» with type «void(struct HWND__ *, Window,  int,  XWindowChanges *)» {aka «void(struct HWND__ *, long unsigned int,  int,  XWindowChanges *)»}
  592 | extern void resize_vk_surfaces( HWND hwnd, Window active, int mask, XWindowChanges *changes ) DECLSPEC_HIDDEN;
      |             ^~~~~~~~~~~~~~~~~~
In file included from ../wine-mirror-git/dlls/winex11.drv/wintab.c:32:
../wine-mirror-git/include/wine/debug.h:501:48: предупреждение: «__wine_dbch___default» определена, но не используется [-Wunused-const-variable=]
  501 |     static struct __wine_debug_channel * const __wine_dbch___default = &__wine_dbch_##ch
      |                                                ^~~~~~~~~~~~~~~~~~~~~
../wine-mirror-git/dlls/winex11.drv/wintab.c:35:1: замечание: в расширении макроса «WINE_DEFAULT_DEBUG_CHANNEL»
   35 | WINE_DEFAULT_DEBUG_CHANNEL(wintab32);
      | ^~~~~~~~~~~~~~~~~~~~~~~~~~
make: *** [Makefile:231874: dlls/winex11.drv/vulkan.o] Ошибка 1
make: *** Ожидание завершения заданий…
 -> Removed BIG_UGLY_FROGMINER - Ribbit
 -> Removed Proton-tkg token - Valve Ribbit
 -> exit cleanup done

**Comment by Tk-Glitch** [comment]
The problem likely comes from the childwindow patch. You can disable it through .cfg to confirm that it fixes the issue. I'm still searching why this doesn't happen to everyone. Maybe a dependency weirdness.

**Comment by pa5hgan** [comment]
Yep, syncing function declaration to version that comes from .h file also helps.

**Comment by Denkevich** [comment]
How to build only 64 bits Wine-TGK?

**Comment by Tk-Glitch** [comment]
> How to build only 64 bits Wine-TKG?

In `wine-tkg-profiles/advanced-customization.cfg`:
```
# Set to true to disable 32-bit wine. Doing so will disable WoW64 and break ALL 32-bit apps. Default is "false".
_NOLIB32="false"
```

**Comment by Curve** [comment]
Got the same issue on arch linux

```
  -> Applying valve_proton_fullscreen_hack-staging.patch
  -> Applying LAA-unix-staging.patch
==> ERROR: Patch application has failed. The error was logged to /home/curve/wine-tkg-git/wine-tkg-git/prepare.log for your convenience.
==> ERROR: A failure occurred in prepare().
    Aborting...
  -> Removed BIG_UGLY_FROGMINER - Ribbit
  -> Removed Proton-tkg token - Valve Ribbit
  -> exit cleanup done
```

```
LAA-unix-staging.patch -- ( Applied large address aware override support )
patching file dlls/kernel32/heap.c
Hunk #2 succeeded at 548 (offset 2 lines).
patching file dlls/ntdll/ntdll.spec
Hunk #1 FAILED at 1617.
1 out of 1 hunk FAILED -- saving rejects to file dlls/ntdll/ntdll.spec.rej
patching file dlls/ntdll/unix/server.c
Hunk #1 succeeded at 1648 (offset 177 lines).
patching file dlls/ntdll/unix/unix_private.h
Hunk #1 succeeded at 519 with fuzz 2 (offset 49 lines).
patching file dlls/ntdll/unix/virtual.c
Hunk #1 succeeded at 3783 (offset 363 lines).
```

**Comment by e666666** [comment]
For the record, installing libvulkan-dev on Debian alike distros appear to fix the issue.

**Comment by Tk-Glitch** [comment]
libvulkan-dev is already part of the dependencies for debian/buntu (https://github.com/Frogging-Family/wine-tkg-git/blob/master/wine-tkg-git/wine-tkg-scripts/deps) covered by the `_nomakepkg_dep_resolution_distro` .cfg option:
```
# Set to the distro of your choice to attempt dependency resolution. Valid options are "debuntu" (for debian, ubuntu and similar), "fedora" or "archlinux".
_nomakepkg_dep_resolution_distro=""
```

For Gentoo or other distros that we don't have a dependency resolution for as of yet, you'll need to install the latest vulkan headers package your distro provides.

@Curve Very different issue, already fixed.

**Comment by Curve** [comment]
> libvulkan-dev is already part of the dependencies for debian/buntu (https://github.com/Frogging-Family/wine-tkg-git/blob/master/wine-tkg-git/wine-tkg-scripts/deps) covered by the `_nomakepkg_dep_resolution_distro` .cfg option:
> ```
> # Set to the distro of your choice to attempt dependency resolution. Valid options are "debuntu" (for debian, ubuntu and similar), "fedora" or "archlinux".
> _nomakepkg_dep_resolution_distro=""
> ```
> 
> For Gentoo or other distros that we don't have a dependency resolution for as of yet, you'll need to install the latest vulkan headers package your distro provides.
> 
> @Curve Very different issue, already fixed.

Yeah my issue's been fixed for a good while now, forgot to mention that, sorry
